src/Workspaces/Core/Portable/CodeGeneration/AbstractFlagsEnumGenerator.cs

```diff
@@ -47,10 +48,11 @@ private static bool IsFlagsEnum(INamedTypeSymbol typeSymbol)
                     var type = ctor.ContainingType;
                     if (!ctor.Parameters.Any() && type.Name == "FlagsAttribute")
                     {
+                        RoslynDebug.Assert(type.ContainingSymbol is not null);
                         var containingSymbol = type.ContainingSymbol;
                         if (containingSymbol.Kind == SymbolKind.Namespace &&
                             containingSymbol.Name == "System" &&
-                            ((INamespaceSymbol)containingSymbol.ContainingSymbol).IsGlobalNamespace)
+                            ((INamespaceSymbol?)containingSymbol.ContainingSymbol)?.IsGlobalNamespace == true)
```

Comment:
I think we could use a `!` here safely. The only `null` containing symbol should be the global namespace, and if we get to this point we've established `containingSymbol` is not the global namespace.